### PR TITLE
Userinterface

### PR DESCRIPTION
**Pull request by PhuocBao**

### CODE REVIEW

File: app/src/main/res/layout/header_navigationdrawer.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    >
+
+    <ImageView
+        android:layout_width="@dimen/dp_280"
+        android:layout_height="@dimen/dp_170"
+        android:contentDescription="@string/content_discription"
+        android:src="@drawable/header_img"
```
Comment (ThuanPx) [inline app/src/main/res/layout/header_navigationdrawer.xml]: đặt tên theo kiểu ic_XXX

File: app/src/main/java/com/framgia/music/ViewPagerAdapter.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import static com.framgia.music.ViewPagerAdapter.TabItem.MYMUSIC_ITEM;
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+
+/**
+ * Created by baohuynh on 22/01/2018.
+ */
+
+public class ViewPagerAdapter extends FragmentPagerAdapter {
+    private Context mContext;
+    private static final int ITEM_COUNT = 2;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/music/ViewPagerAdapter.java]: 2.8 Class member ordering
This is not the only correct solution, but it is recommended to use the following order :

Constants
Fields
Constructors
Override methods and callbacks (public or private)
Public methods
Private methods
Inner classes or interfaces

https://github.com/framgia/coding-standards/blob/master/eng/android/codingstyleguide.md

File: app/src/main/java/com/framgia/music/MainActivity.java [hunk 3]
```diff
@@ -1,13 +1,65 @@
...
+    }
+
+    private void initToolbar() {
+        setSupportActionBar(mToolbar);
+        setTitle(R.string.title);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+    }
+
+    private void initViewPager() {
+        ViewPager mViewPager = findViewById(R.id.viewpager_main);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/music/MainActivity.java]: biến này chỉ cần đặt là viewPager laà đuuowcj

File: app/src/main/java/com/framgia/music/MainActivity.java [hunk 4]
```diff
@@ -1,13 +1,65 @@
...
+        ViewPagerAdapter adapter = new ViewPagerAdapter(getSupportFragmentManager(), this);
+        mViewPager.setAdapter(adapter);
+        mTabLayout.setupWithViewPager(mViewPager);
+    }
+
+    private void initDrawerNavigation() {
+        NavigationView mNavigationView = findViewById(R.id.navigation_main);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/music/MainActivity.java]: tương tự

File: app/src/main/java/com/framgia/music/MainActivity.java [hunk 5]
```diff
@@ -1,13 +1,65 @@
...
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.view.MenuItem;
 
 public class MainActivity extends AppCompatActivity {
+    private Toolbar mToolbar;
+    private TabLayout mTabLayout;
+    private DrawerLayout mDrawerLayout;
+    private ActionBarDrawerToggle mActionBarDrawerToggle;
+    private static final String API_KEY = BuildConfig.API_KEY;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/music/MainActivity.java]: 2.8 Class member ordering
This is not the only correct solution, but it is recommended to use the following order :

Constants
Fields
Constructors
Override methods and callbacks (public or private)
Public methods
Private methods
Inner classes or interfaces

https://github.com/framgia/coding-standards/blob/master/eng/android/codingstyleguide.md

File: app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -1,21 +1,42 @@
...
     xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/drawerlayout_main"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     tools:context="com.framgia.music.MainActivity"
     >
 
-    <TextView
-        android:layout_width="wrap_content"
+    <LinearLayout
```
Comment (DoanVanToan) [inline app/src/main/res/layout/activity_main.xml]: Sao k dùng constraint layout em?

File: app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -1,21 +1,42 @@
...
-        />
+        android:orientation="vertical"
+        >
+        <include
+            layout="@layout/toolbar"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            />
+
+        <LinearLayout
```
Comment (DoanVanToan) [inline app/src/main/res/layout/activity_main.xml]: remove thẻ LinearLayout, k cần thiết

File: app/src/main/res/layout/toolbar.xml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        app:titleTextColor="@color/black"
+        />
+
+    <android.support.design.widget.TabLayout
+        android:id="@+id/tablayout_main"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@color/colorToolbar"
+        app:tabGravity="fill"
+        app:tabIndicatorColor="@color/white"
```
Comment (DoanVanToan) [inline app/src/main/res/layout/toolbar.xml]: e có thể sử dung android:color/white, black mà k cần định nghĩa thêm. Android đã định nghĩa sẵn

File: app/src/main/res/values/colors.xml [hunk 9]
```diff
@@ -1,6 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
     <color name="colorPrimary">#3F51B5</color>
-    <color name="colorPrimaryDark">#303F9F</color>
+    <color name="colorPrimaryDark">#7aec71</color>
     <color name="colorAccent">#FF4081</color>
+    <color name="colorToolbar">#56df6f</color>
```
Comment (DoanVanToan) [inline app/src/main/res/values/colors.xml]: e check lại convention về naming resource
3.2 Naming resources
Resource IDs and names need to be declared in lowercase_underscore

Comment (DoanVanToan) [reply app/src/main/res/values/colors.xml]: color e nên đặt là color_<color_name>

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -1,3 +1,17 @@
 <resources>
     <string name="app_name">IMusic</string>
+
+    <!-- TODO: Remove or change this placeholder text -->
```
Comment (DoanVanToan) [inline app/src/main/res/values/strings.xml]: remove comment

**Pull request merged**